File: Day_17/php/the_class_set.php
<?php

class User {
    private $data = [];

    public function __set($name, $value)
    {
        $this->data[$name] = $value;
    }

    public function __get($name)
    {
        if (array_key_exists($name, $this->data)) {
            return $this->data[$name];
        }
        return null; // You can return any default value you prefer
    }

    public function save()
    {
        return $this->data;
    }
}

// using __set
$user = new User();
$user->name = 'John Doe';
$user->email = 'john@example.com';
$user->password = 'password';

// Accessing the properties using __get
echo $user->name . "\n";
echo $user->email . "\n";
echo $user->password . "\n";

$result = $user->save();

// Output the result as an array
print_r($result);

File: Day_17/Python/the_class_set.py
import bcrypt

class User:
    def __init__(self):
        self.data = {}

    def __setattr__(self, name, value):
        # Store data in the dictionary, but avoid infinite recursion
        print(name)
        if name != 'data':
            self.data[name] = value
        else:
            # Call the base class's __setattr__ for 'data' attribute
            super().__setattr__(name, value)

    def __getattr__(self, name):
        return self.data.get(name, None)

    def save(self):
        return self.data

# Usage
user = User()
user.name = 'John Doe'
user.age = 30

# Accessing the properties using __getattr__
print(user.name)  # Output: John Doe
print(user.age)   # Output: 30

result = user.save()

# Output the result as a dictionary
print(result)
